src/app/ngrx/reducers/episode-alltime-downloads.reducer.ts

```diff
@@ -4,12 +4,20 @@ import { AllActions, ActionTypes } from '../actions';
 
 export type State = EntityState<EpisodeAllTimeDownloads>;
 
-export const adapter: EntityAdapter<EpisodeAllTimeDownloads> = createEntityAdapter<EpisodeAllTimeDownloads>({
-  selectId: (e: EpisodeAllTimeDownloads) => e.guid,
-});
+export const adapter: EntityAdapter<EpisodeAllTimeDownloads> = createEntityAdapter<EpisodeAllTimeDownloads>();
 
 export const initialState: State = adapter.getInitialState({});
 
+export const {
```

Comment:
You'll see that I moved these declarations above the reducer in a lot of places where I wanted to use them inside the reducer to select existing entity records.

Reply:
Speaking of destructuring renaming, I like using it on the results of `getSelectors()` to simplify things a bit here. It looks like we're exporting both the original and aliased names for these particular selectors too, which might lead to some confusion in the future (obviously if both are used and we need to preserve this for now that's a different story):

```
export const {
  selectIds: selectEpisodeAllTimeDownloadsGuids,
  selectEntities: selectEpisodeAllTimeDownloadsEntities,
  selectAll: selectAllEpisodeAllTimeDownloads,
} = adapter.getSelectors();
```

Reply:
Good call. My earlier pattern was just following what I'd learned in beginner tutorials, but I agree, there's good reason to go this route.